drv/lpc55-swd/src/main.rs

```diff
@@ -1534,7 +1561,7 @@ impl ServerImpl {
                     limit -= 1;
                     hl::sleep_for(1);
                 }
-                Err(_) => return Err(SpCtrlError::Fault),
+                Err(e) => return Err(e.into()),
```

Comment:
similarly, if we don't mind the much bigger diff, the whole `match` could just be:
```rust
let dhcsr = self.dp_read_bitflags::<Dhcsr>()?;
if dhcsr.is_regrdy() {
    // Trace retries used
    if limit != RETRY_LIMIT {
        ringbuf_entry!(Trace::LimitRemaining(limit));
    }
    return Ok(());
}
if limit == 0 {
    ringbuf_entry!(Trace::LimitRemaining(limit));
    return Err(SpCtrlError::Fault);
}
limit -= 1;
hl::sleep_for(1);
```

Reply:
Good suggestion on both paths

Reply:
This inspired me to make similar changes elsewhere. I think it cleans up the code nicely!